### PR TITLE
Add dataset configuration example

### LINKED ISSUE #55: Add or link documentation for TSML
The README describes how to start the server and how to point it to TSML files, but not how to make TSML files. Either point to existing docs or add some. Perhaps a worked example?

### PR DESCRIPTION
**Pull request by lindholc**
I've put together an example describing how to write TSML (at a very high level) for a simple example dataset.

- Is this a reasonable example?
- Should I do something more complex?
- Did I say anything wrong?
- Is there something I could explain better?

I wanted to have *something* that someone could look at until we have LaTiS 3, FDML, and tooling for FDML in place.

I recommend reading this document [here](https://github.com/lasp/hapi-server/blob/tsml-guide/docs/configuring-datasets.md) so the markdown is rendered.

Closes #55.

### CODE REVIEW

**Comment by lindholc** [comment]
@pfisterp, I'd rather link to the schema than duplicate it here. We could probably put it in the LaTiS 2 repo and then link to it from this document.

**Comment by dlindhol** [comment]
We should consider a master latis site/service where we can host the schema file via http so fdml files can use that for the `schemaLocation`.

File: docs/configuring-datasets.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+<?xml version="1.0" encoding="UTF-8"?>
+
+<dataset id="scalar">
+
+    <adapter class="latis.reader.tsml.AsciiAdapter"
+             location="${dataset.dir}/scalar.csv"
+             delimiter=","
+             commentCharacter="#"/>
+
+    <!-- We've described the shape of the data. -->
```
[the remark this answers is not in the document]
Comment (lindholc) [reply docs/configuring-datasets.md]: I have a link to the rendered document in the PR description.

I don't think there's a way to highlight changes in the code blocks.

File: docs/configuring-datasets.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        <!-- There is still more to add here. -->
+    </integer>
+
+</dataset>
+```
+
+The `time` tag adds a time variable, and the `type` attribute
+specifies that the time is formatted text.
+
+The `integer` tag adds a variable that is an integer, and the `id`
```
[the remark this answers is not in the document]
Comment (lindholc) [reply docs/configuring-datasets.md]: I don't think so. I'd rather reword things if you're worried people might thing we mean a literal machine integer.

File: docs/configuring-datasets.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+The other attributes depend on the adapter being used. We've set the
+following for the ASCII adapter:
+
+- `location` : Specifies the location of our CSV file.
+  - Note that `${dataset.dir}` is expanded to where the server is
+    configured to look for datasets. (See the [README][readme] for
+    those configuration options.)
+- `delimiter` : The character that separates data values/columns.
+- `commentCharacter` : The character that indicates header/comment
+  lines.
```
[the remark this answers is not in the document]
Comment (lindholc) [reply docs/configuring-datasets.md]: I'd like to think people would understand that comment lines would be skipped.

File: docs/configuring-datasets.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+        <metadata units="units"/>
+    </integer>
+
+</dataset>
+```
+
+A `metadata` tag adds the metadata specified by its attributes:
+
+- `units` : Adds units to a variable.
+  - For times represented by formatted text, this is a time format
```
[the remark this answers is not in the document]
Comment (lindholc) [reply docs/configuring-datasets.md]: Done.

File: docs/configuring-datasets.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+terms of that model.
+
+The gist of the functional data model is that we have three types of
+things to work with:
+
+- *scalars* : A single variable. (e.g. time, flux, temperature)
+- *tuples* : A collection of variables. (e.g. a wind vector, a pair of
+  flux and temperature)
+- *functions* : A mapping from a domain (independent) variable to a
+  codomain (dependent) variable. (e.g. temperature as a function of
```
[the remark this answers is not in the document]
Comment (lindholc) [reply docs/configuring-datasets.md]: Done.

File: docs/configuring-datasets.md [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+```
+
+A `metadata` tag adds the metadata specified by its attributes:
+
+- `units` : Adds units to a variable.
+  - For times represented by formatted text, this is a time format
+    string. This is so LaTiS can read and convert times. The server
+    will report "UTC" as the units as required by the HAPI
+    specification.
+- `length` : Adds the length of text variables.
```
[the remark this answers is not in the document]
Comment (lindholc) [reply docs/configuring-datasets.md]: I'm going to hope that we don't.

File: docs/configuring-datasets.md [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+2018-01-01T00:00:01Z,1
+2018-01-01T00:00:02Z,2
+```
+
+We will need to provide a TSML file that describes this dataset.
+
+### Introducing a Dataset
+
+First, here's an example of a bare-bones TSML file for a dataset we
+will call `scalar`:
```
[the remark this answers is not in the document]
Comment (lindholc) [reply docs/configuring-datasets.md]: I've renamed everything to `simple_dataset`.

**Comment by lindholc** [comment]
I've pushed some updates to address comments. Part of these updates include having an `examples` directory that includes these files. As a reminder, you can see the rendered markdown [here](https://github.com/lasp/hapi-server/blob/tsml-guide/docs/configuring-datasets.md).

Regarding a `log` dataset: I wouldn't want to include it by default and have people accidentally exposing their logs. I don't even know that I would want to include it as an example and potentially have people try to use it.

**Pull request merged**